### ISSUE TITLE
[All Platforms] Visual difference for MaterialCheckBoxStyle with latest Uno version

### ISSUE THREAD

**Issue by agneszitte** [unoplatform/uno#13464]
### Current behavior

Visual difference for MaterialCheckBoxStyle with the latest Uno version (Tested with Uno 5.0.0-dev.2319 version)
Note that for the screenshots taken here, there was no platform-specific code involved (I removed the xamarin specific margin for CheckBox for example while testing and taking the screenshots to have a better idea of the actual result for all platforms)

[Android]
![image](https://github.com/unoplatform/uno/assets/16295702/8754509f-5805-446f-8309-12620cecc7d3)

[Skia.GTK]
![image](https://github.com/unoplatform/uno/assets/16295702/1daddd4d-2917-475f-a683-d6aa137d7dae)

[WASM]
![image](https://github.com/unoplatform/uno/assets/16295702/09d0a07e-3567-4bff-a973-2251925a88d5)

[iOS] ?
Current exception preventing to test iOS
```
System.InvalidCastException: 'Unable to cast object of type 'Foundation.NSObject' (Objective-C type: 'Uno_UI_Controls_UnoNavigationBar') to type 'Uno.UI.Controls.UnoNavigationBar'.
Additional information:
    Selector: setFrame:
    Method: Uno.UI.Controls.UnoNavigationBar:set_Frame (CoreGraphics.CGRect)
'
```

### Expected behavior

MaterialCheckBoxStyle should look like Windows on all platforms
[Windows]
![image](https://github.com/unoplatform/uno/assets/16295702/af1edfa2-c519-4475-b958-b05f2d0388b3)

### How to reproduce it (as minimally and precisely as possible)

- Launch Uno.Themes sample App on all platforms
- Take a look at the CheckBox sample on the main OverviewPage
- Notice the difference in the placement of the glyphs inside (checked & indeterminate) for the different platforms compared to Windows

### Workaround

N/A

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

Uno.UI Uno 5.0.0-dev.2319

### Affected platforms

WebAssembly, Android, Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022

### IDE version

Version 17.6.4

**Comment by sakshi173** [comment]
@agneszitte,As suggested below is the compared screen shot for 5X and 4X.
**WASM**
![checkbox wasm](https://github.com/unoplatform/uno/assets/74563602/8d6f5d60-4176-4e9e-b9e4-d52f8905e738)
![checkbox iOS](https://github.com/unoplatform/uno/assets/74563602/63c4592e-33b2-45ec-8bd6-f9062434b12d)
![checkbox Android](https://github.com/unoplatform/uno/assets/74563602/f2e09cb6-fa58-4945-a2b2-eccd94ab8a33)

**Comment by agneszitte** [comment]
@Youssef1313 I don't think this one is a scaling issue like https://github.com/unoplatform/uno/issues/13465#issuecomment-1703097395

**Comment by Youssef1313** [comment]
I checked on Skia, the problem is in the arrangement of the glyph (the glyph is moving down an extra pixel). The measuring looks correct to me.

**Comment by Youssef1313** [comment]
**Testing with the code in the previous comment..**

#### Uno 4.9.45 (latest stable)

##### Skia.Gtk:

CheckGlyph is measured incorrectly, and not centered both horizontally and vertically.

##### Android (Tablet M-DPI 10.1in)

CheckGlyph is measured incorrectly, and not centered both horizontally and vertically.

##### Android (Tablet H-DPI 13.5in)

Not centered both horizontally and vertically.

#### Uno 5.0.0-dev.2382

##### Skia.Gtk:

Measured correctly, cenetered horizontally but not vertically (it is one pixel down than it should)

##### Android (Tablet M-DPI 10.1in)

Measured correctly, and centered both horizontally and vertically.

##### Android (Tablet H-DPI 13.5in)

Centered vertically, but not horizontally.

**Comment by Youssef1313** [comment]
https://github.com/unoplatform/uno/pull/13391 might improve this on Skia.

**Comment by Youssef1313** [comment]
Skia & Wasm might get fixed with 5.0.0-dev.2533

**Comment by Youssef1313** [comment]
For Android and iOS, it will be hard to get this right without fixing the way layouting works on those platforms. cc @jeromelaban @carldebilly

**Comment by Youssef1313** [comment]
What we can do is adjust the margin for the material checkbox glyph specifically. But that doesn't fix the root issue.

**Comment by MartinZikmund** [comment]
No longer relevant for Skia targets, where it works correctly
